### DISCUSSION TITLE
Anyone know how to execute dynamic SQL in codewars' postgres server?

### DISCUSSION THREAD

**Discussion by ChairmanMawh** [codewars/codewars.com#3047] [Q&A]
Just curious, if I had an sql like:

```
SELECT 'SELECT 1 as "' || TO_CHAR(CURRENT_DATE(), 'FMDYDDMONYY') ||'"' as Q
```

Which today would produce a resultset like:

```
Q
-
SELECT 1 as "THU24AUG23"
```

How do I then immediately execute this dynamically generated SQL, to form the actual output?

i.e. How would I execute the original query it in the codewards SQL panel, such that the output was immediately this (today):

```
THU24AUG23
----------
1
```

And tomorrow the same original query output would be:

```
FRI25AUG23
----------
1
```

And so on?

i.e. If I write an SQL whose output is a single string containing a valid SQL, how do I execute the query inside that string and retrieve its output as the output of the challenge?

**Comment by jessya020** [comment]
To execute the dynamically generated SQL string (which your query produces) and get its output directly, you need to use dynamic SQL functionality (like EXECUTE IMMEDIATE in many SQL dialects).
First, your original query generates a string (e.g., SELECT 1 as "THU24AUG23"). Then, you’d take that string and pass it to a dynamic SQL command—this command runs the generated SQL as a live query, and returns its result (in this case, the value 1 under the date-based column name).
In Codewars’ SQL panel, you’d structure your solution to first generate the SQL string, then use the platform’s supported dynamic SQL method to execute that string and return its output. The exact approach depends on the SQL dialect Codewars uses for the kata, but the core idea is: generate the SQL text, then execute that text dynamically.
